### PR TITLE
Added Employee Weekly Timecard Report

### PR DESCRIPTION
**Pull request by shahmharsh**
- Things remaining:
  a) Report still needs to show total number of hours clocked by each employee
  b) Adding title and beautification

### CODE REVIEW

**Comment by shahmharsh** [comment]
@raincoatforever

File: server.js [hunk 1]
```diff
@@ -5,34 +5,37 @@ var app = express();
...
+var COLOR_DEFAULT = "#ffffff";
+var COLOR_CARPETER = "#66ffff";
+var COLOR_MASON = "#ffff00";
+var COLOR_LABORER = "#66ff66";
```
Comment (raincoatforever) [inline server.js]: maybe 

```
var COLOR = {}
COLOR.default = ...
COLOR.carpenter = ...
.
.
```

so `carpenter` in `trade` links automatically to `COLOR.trade`

File: server.js [hunk 2]
```diff
@@ -84,96 +81,134 @@ function GenerateDispatchBoardReport(jsonObj, res) {
...
+                console.log(pdf.numberOfPages);
+                res.setHeader("Content-Type", "application/pdf");
+                pdf.stream.pipe(res);
+            });
+        }
+    );
+}
+
+//Returns employee colors code according to trade
+function getEmployeeColorCode(employee) {
```
Comment (raincoatforever) [inline server.js:189]: if we implement the above soln, then this gets simplified

File: server.js [hunk 3]
```diff
@@ -84,96 +91,122 @@ function GenerateDispatchBoardReport(jsonObj, res) {
...
+
+            conversion(opt, function(err, pdf) {
+                response.setHeader("Content-Type", "application/pdf");
+                pdf.stream.pipe(response);
             });
+        });
+}
 
+function getJobHours(job) {
+    var days = ['sat', 'sun', 'mon', 'tue', 'wed', 'thu', 'fri'];
```
Comment (raincoatforever) [inline server.js:134]: declare static

File: server.js [hunk 4]
```diff
@@ -84,96 +91,122 @@ function GenerateDispatchBoardReport(jsonObj, res) {
...
 
-    jsonObj.rdata.header_color = "#6495ed";
-    // convert array to map-array
-        jsonObj.rdata.data.datesArray = function() {
-            return this.dates.map(function (date) {
-                return {src: date};
-            });
-        };
+function GenerateReportEmployeeWeeklyTimecard(jsonObj, res) {
+    for (var i = 0; i < jsonObj.rdata.data.employees.length; i++) {
```
Comment (raincoatforever) [inline server.js:157]: will die if employees == undefined

**Comment by raincoatforever** [comment]
:+1:

File: server.js [hunk 5]
```diff
@@ -5,34 +5,49 @@ var app = express();
...
+var WEEKLY_DISPATCH_HEADER = "weeklyDispatchHeader";
+var WEEKLY_TIMECARD_HEADER = "weeklyTimecardHeader";
+
+var COLOR = {};
+COLOR[CARPENTER] = "#66ffff";
+COLOR[MASON] = "#ffff00";
+COLOR[LABORER] = "#66ff66";
+COLOR[DEFAULT] = "#ffffff";
+COLOR[WEEKLY_DISPATCH_HEADER] = "#6495ed";
+COLOR[WEEKLY_TIMECARD_HEADER] = "#bdbdbd";
```
Comment (raincoatforever) [inline server.js:31]: good !

File: server.js [hunk 6]
```diff
@@ -69,111 +76,147 @@ function createHTMLFromJSON(jsonData, res) {
...
-
-    console.log(JSON.stringify(jsonObj));
-    console.log(jsonObj.rdata.data);
-
-    for( var i=0; i < jsonObj.rdata.data.length; i++){
+function generateDispatchBoardReport(jsonObj, res) {
+    for( var i=0; i < jsonObj.rdata.data.length; i++) {
 
         var dataNode = jsonObj.rdata.data[i];
         console.log(JSON.stringify(dataNode.site));
```
Comment (raincoatforever) [inline server.js]: remove this

File: templates/EmployeeWeeklyTimecard.html [hunk 7]
```diff
@@ -0,0 +1,202 @@
+<html>
```
Comment (raincoatforever) [inline templates/EmployeeWeeklyTimecard.html:1]: attach a screenshot of the report

**Comment by raincoatforever** [comment]
rest looks good !

**Comment by shahmharsh** [comment]
screen shot 2015-07-14 at 9 19 03 pm

**Pull request merged**